editor/src/components/inspector/flex-section.tsx

```diff
@@ -457,59 +457,92 @@ const TemplateDimensionControl = React.memo(
-                  visibility: 'hidden',
-                },
-                ':hover': {
-                  [`& > .${axisDropdownMenuButton}`]: {
-                    visibility: 'visible',
-                  },
-                },
-              }}
-            >
-              <div style={{ display: 'flex', alignItems: 'center', gap: 6, flex: 1 }}>
-                <Subdued
-                  style={{
-                    width: 40,
-                    overflow: 'hidden',
-                    textOverflow: 'ellipsis',
-                    whiteSpace: 'nowrap',
-                  }}
-                  title={value.areaName ?? undefined}
-                >
-                  {value.areaName ?? index + 1}
-                </Subdued>
-                <NumberOrKeywordControl
-                  style={{ flex: 1 }}
-                  testId={testId}
-                  value={value.value}
-                  keywords={gridDimensionDropdownKeywords}
-                  keywordTypeCheck={isValidGridDimensionKeyword}
-                  onSubmitValue={onUpdate(index)}
-                  controlStatus={
-                    isGridCSSKeyword(value) && value.value.value === 'auto' ? 'off' : undefined
-                  }
-                />
-              </div>
-              <SquareButton className={axisDropdownMenuButton}>
-                <DropdownMenu align='end' items={dropdownMenuItems(index)} opener={openDropdown} />
-              </SquareButton>
-            </div>
-          )
-        })}
+        {values.map((value, index) => (
+          <AxisDimensionControl
+            key={index}
+            value={value}
+            index={index}
+            axis={axis}
+            onUpdate={onUpdate}
+            items={dropdownMenuItems(index)}
+            opener={openDropdown}
+          />
+        ))}
       </div>
     )
   },
 )
 TemplateDimensionControl.displayName = 'TemplateDimensionControl'
 
+function AxisDimensionControl({
+  value,
+  index,
+  items,
+  axis,
+  onUpdate,
+  opener,
+}: {
+  value: GridDimension
+  index: number
+  items: DropdownMenuItem[]
+  axis: 'column' | 'row'
+  onUpdate: (
+    index: number,
+  ) => (value: UnknownOrEmptyInput<CSSNumber | CSSKeyword<ValidGridDimensionKeyword>>) => void
+  opener: (isOpen: boolean) => React.ReactElement
+}) {
+  const testId = `grid-dimension-${axis}-${index}`
+  const [isOpen, setIsOpen] = React.useState(false)
+  const onOpenChange = React.useCallback((isDropdownOpen: boolean) => {
+    setIsOpen(isDropdownOpen)
+  }, [])
+  return (
+    <div
+      key={`col-${value}-${index}`}
+      style={{ display: 'flex', alignItems: 'center', gap: 6 }}
+      className={isOpen ? 'openMenu' : ''}
+      css={{
+        [`& > .${axisDropdownMenuButton}`]: {
+          visibility: 'hidden',
+        },
+        ':hover, &.openMenu': {
+          [`& > .${axisDropdownMenuButton}`]: {
+            visibility: 'visible',
+          },
+        },
+      }}
```

Comment:
Why not use the `open` variable instead of the classname-based code here?

Reply:
i was debating myself on that. 
to keep the css specificity logic we have to do it here (using the `open` variable in the `style` prop will be overriding this definition.
so the other way I considered was:
```tsx
css={{
  [`& > .${axisDropdownMenuButton}`]: {
    visibility: isOpen ? 'visible' : 'hidden', // using it here
  },
  ':hover': {
    [`& > .${axisDropdownMenuButton}`]: {
      visibility: 'visible',
    },
  },
}}
```
but this felt a bit clunky and also not very extensible if we'd want other cases. also if in the future we will want another thing to happen, we will have to change both places.

I thought that adding it as another case in the selector next to the `:hover` class makes most sense, since it really is "either it is hovered or the menu is opened".
We can also use a data-attribute instead of a class (`data-dropdown-open`), but I would still add it to the `:hover` css selector, imo.